go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -82,45 +84,140 @@ func (ovn *Controller) getLoadBalancerVIPS(
+	klog.V(5).Infof("Creating lb with %s, %s, %d, [%v], %d", lb, serviceIP, port, ips, targetPort)
 
 	var commaSeparatedEndpoints string
-	for i, ep := range ips {
-		comma := ","
-		if i == 0 {
-			comma = ""
+
+	if len(ips) > 0 {
+		for i, ep := range ips {
+			comma := ","
+			if i == 0 {
+				comma = ""
+			}
+			commaSeparatedEndpoints += fmt.Sprintf("%s%s", comma, util.JoinHostPortInt32(ep, targetPort))
 		}
-		commaSeparatedEndpoints += fmt.Sprintf("%s%s", comma, util.JoinHostPortInt32(ep, targetPort))
+
+		// ensure the ACL is removed if it exists
+		ovn.deleteLoadBalancerRejectACL(lb, util.JoinHostPortInt32(serviceIP, port))
 	}
-	target := fmt.Sprintf(`vips:"%s"="%s"`, util.JoinHostPortInt32(serviceIP, port), commaSeparatedEndpoints)
 
-	out, stderr, err := util.RunOVNNbctl("set", "load_balancer", lb,
-		target)
+	return ovn.configureLoadBalancer(lb, serviceIP, port, commaSeparatedEndpoints)
+}
+
+func (ovn *Controller) getLogicalSwitchesForLoadBalancer(lb string) ([]string, error) {
+	out, _, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=_uuid", "find",
+		"Logical_Switch", fmt.Sprintf("load_balancer{>=}%s", lb))
 	if err != nil {
-		klog.Errorf("Error in creating load balancer: %s "+
-			"stdout: %q, stderr: %q, error: %v", lb, out, stderr, err)
+		return nil, err
 	}
-	return err
+	return strings.Fields(out), nil
+}
+
+func (ovn *Controller) createLoadBalancerRejectACL(lb string, serviceIP string, port int32, proto kapi.Protocol) (string, error) {
+
+	switches, err := ovn.getLogicalSwitchesForLoadBalancer(lb)
+	if err != nil {
+		klog.Errorf("Error finding logical switch that contains load balancer %s", lb)
+		return "", err
+	}
+
+	if len(switches) == 0 {
+		klog.Infof("Ignoring creating reject ACL for load balancer %s. It has no logical switches", lb)
+		return "", nil
+	}
+
+	ip := net.ParseIP(serviceIP)
+	if ip == nil {
+		klog.Errorf("Cannot parse IP address %s", serviceIP)
+		return "", fmt.Errorf("cannot create reject ACL, invalid cluster IP: %s", serviceIP)
+	}
+	var aclMatch string
+	var l3Prefix string
+	if ip.To4() != nil {
+		l3Prefix = "ip4"
+	} else {
+		l3Prefix = "ip6"
+	}
+	vip := util.JoinHostPortInt32(serviceIP, port)
+	if eps, ok := ovn.GetServiceLBEndpoints(lb, vip); ok && len(eps) > 0 {
+		klog.Warningf("ACL being applied to Load Balancer with Endpoints: %s, %s", lb, vip)
+	}
+
+	aclMatch = fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, serviceIP,
+		strings.ToLower(string(proto)), strings.ToLower(string(proto)), port)
+
+	cmd := []string{"--id=@acl", "create", "acl", "direction=from-lport", "priority=1000", aclMatch, "action=reject"}
+	for _, ls := range switches {
+		cmd = append(cmd, "--", "add", "logical_switch", ls, "acls", "@acl")
+	}
+
+	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
+	if err != nil {
+		klog.Errorf("Error creating ACL reject rule: %s for load balancer %s: %s", cmd, lb, stderr)
+		return "", err
```

Comment:
@trozet let's pass descriptive errors up to the callers to log (if they wish) rather than logging in each function *and* passing the error up.

Reply:
will fix